### DISCUSSION TITLE
Quota Errors

### DISCUSSION THREAD

**Discussion by addohm** [abraunegg/onedrive#2658] [Q&A] [answered]
I see several issues, commits and discussions regarding this but no really specific resolutions.  What is causing this and what can be done to fix it?  This account has 5TB and is only using 256 GB.

```
  API Request ID:   3d628eaa-a2f4-44da-819c-966efb6e27fd
Uploading new file ./pinas_Backups/docker/pinas_docker_backup.7z ... 
skipped.
ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 507 ()
  Error Reason:     Quota limit reached
  Error Timestamp:  2024-03-07T23:29:15
  API Request ID:   74e0f9fb-9d6c-4db2-9f62-067293c880e8
WARNING: OneDrive quota information is being restricted or providing a zero value. Please fix by speaking to your OneDrive / Office 365 Administrator.
```

**Comment by abraunegg** [comment, marked as answer]
@addohm 
Whilst you *may think* you have 5TB, you might not actually have that available space online.

When you encounter a 507 Insufficient Storage error response from the Microsoft Graph API with the reason being "Quota limit reached," it indicates that your account has exceeded the storage quota allocated by Microsoft for your specific resource or service.

To resolve this error, consider the following steps:

**Review Storage Usage:** Identify which service has exceeded its quota. Microsoft provides tools and settings in the Admin Center to review and manage storage. Empty online recycle bin for example as this *contributes* to quota usage.
**Increase Storage Quota:** If possible, increase the storage quota for the affected service. This may involve adjusting settings within the Microsoft 365 Admin Center or purchasing additional storage from Microsoft.
**Clean Up Resources:** As per above, identify and delete unnecessary files, emails, or data that are consuming excessive storage. Implementing retention policies or archiving strategies can help manage storage effectively.

In addition to this, I *strongly* advise in your environment, you build a custom Docker image from v2.5.0 'alpha-5' to make use of all the code change, quota tracking change & file replacement / update change - specifically when dealing with Microsoft OneDrive Business accounts.

**Reply by addohm** [reply]
I think it was probably the trash as I've been cycling data in and out repeatedly the past week.  Clearing out and trying again.

I appreciate your work on this and the next version.  I will certainly move one of my containers to alpha when I have more flexibility.

Edit: Confirmed, recycle bin was the culprit.

**Reply by abraunegg** [reply]
Glad that was the issue.

If and *when* you rebuild using 'alpha-5', please ensure you read carefully the *updated* documentation. Many things change with v2.5.x:

(from the latest PR that will be rolled into 'alpha-5')

* https://github.com/abraunegg/onedrive/blob/Implement-Feature-Request-1300/docs/application-security.md
* https://github.com/abraunegg/onedrive/blob/Implement-Feature-Request-1300/docs/business-shared-items.md
* https://github.com/abraunegg/onedrive/blob/Implement-Feature-Request-1300/docs/client-architecture.md
* https://github.com/abraunegg/onedrive/blob/Implement-Feature-Request-1300/docs/usage.md
* https://github.com/abraunegg/onedrive/blob/Implement-Feature-Request-1300/docs/application-config-options.md
